x/logic/predicate/uri.go

```diff
@@ -155,39 +152,39 @@ func (comp Component) Unescape(v string) (string, error) {
 //
 // [RFC 3986]: https://datatracker.ietf.org/doc/html/rfc3986#section-2.1
 func URIEncoded(vm *engine.VM, component, decoded, encoded engine.Term, cont engine.Cont, env *engine.Env) *engine.Promise {
-	return engine.Delay(func(ctx context.Context) *engine.Promise {
-		var comp Component
-		switch c := env.Resolve(component).(type) {
-		case engine.Atom:
-			cc, err := NewComponent(c.String())
-			if err != nil {
-				return engine.Error(fmt.Errorf("uri_encoded/3: %w", err))
-			}
-			comp = cc
-		default:
-			return engine.Error(fmt.Errorf("uri_encoded/3: invalid component type: %T, should be Atom", component))
+	var comp Component
+	switch c := env.Resolve(component).(type) {
+	case engine.Atom:
+		cc, err := NewComponent(c.String())
+		if err != nil {
+			return engine.Error(fmt.Errorf("uri_encoded/3: %w", err))
 		}
+		comp = cc
+	default:
+		return engine.Error(fmt.Errorf("uri_encoded/3: invalid component type: %T, should be Atom", component))
```

Comment:
The `uri_encoded` predicate seems to have escaped the refactoring 👀

Reply:
Yes true. I missed that one :sweat_smile: 

Ideally, we should use [ashanbrown/forbidigo](https://github.com/ashanbrown/forbidigo) to avoid this. But it takes time to configure it correctly.

Reply:
@amimart done with b8d8133b4a2c20f11831b3e441f26b588a7e7776.